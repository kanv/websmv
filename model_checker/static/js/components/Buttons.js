import React from 'react'

export class Buttons extends React.Component {

	constructor(props) {
		super(props);
		this.onButtonClick = this.onButtonClick.bind(this);
	}

	static propTypes = {
        active: React.PropTypes.string.isRequired,
        callback: React.PropTypes.func.isRequired
	};

	onButtonClick(e) {
		this.props.callback(e.target.value);
	}

	render() {
		let _this = this;
		let active;
		if (_this.props.active == 'sim') {
		    active = ['button-list-item-active', 'button-list-item-inactive'];
        } else if (_this.props.active == 'tl') {
		    active = ['button-list-item-inactive', 'button-list-item-active'];
        }

		return (
			<ul className="button-list">
                <li key="sim" className="button-list-item">
                    <button value="sim" className={active[0]} onClick={_this.onButtonClick}>
                    Simulate
                    </button>
                </li>
                <li key="tl" className="button-list-item">
                    <button value="tl" className={active[1]} onClick={_this.onButtonClick}>
                    Verify
                    </button>
                </li>
			</ul>
        );
	}
}
