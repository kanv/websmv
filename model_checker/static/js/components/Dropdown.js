import React from 'react'

export class Dropdown extends React.Component {
	
	// constructor() {
	// 	super()
	// 	this.onButtonClick = this.onButtonClick.bind(this)
	// }

	// static propTypes = {

	// }

	// onButtonClick(e) {
	// 	this.props.toggleActive(e.target.id.toLowerCase())
	// }

	render() {
		let _this = this
		return (
			<div class="dropdown">
		  		<button class="dropbtn">Dropdown</button>
			  		<div class="dropdown-content">
			    		<a href="#">Link 1</a>
			    		<a href="#">Link 2</a>
			    		<a href="#">Link 3</a>
			  		</div>
			</div>);
	}
}
