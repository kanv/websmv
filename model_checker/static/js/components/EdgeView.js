import React from 'react'
import findObjInArray from '../util/ArrayAdditions'

export class EdgeView extends React.Component {

    static propTypes = {
        edge: React.PropTypes.object.isRequired
    };

    render() {
		let _this= this;
		return(
			<ul className="state-list">
				{
					_this.props.edge.value.map((value)=>{
						return(
							<li key={value.name}>
							<label>{value.name}</label>
							<div/>
							<label>{value.value}</label>
							</li>
						)
					})
				}
			</ul>
		)
    };
}
