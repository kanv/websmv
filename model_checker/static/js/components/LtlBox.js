import React from 'react'

export class LtlBox extends React.Component {

    constructor(props) {
        super(props);
        this.selected = '';
        this.binary = ['and', 'or', 'not', '()', '->', '<->'];
        this.ctl = ['G','F','X','U'];
        this.addState = this.addState.bind(this);
        this.changeFormula = this.changeFormula.bind(this);
        this.onAppend = this.onAppend.bind(this);
        this.changeState = this.changeState.bind(this);
        this.addLtl = this.addLtl.bind(this);
    }

	static propTypes = {
        formula: React.PropTypes.string.isRequired,
        states: React.PropTypes.array.isRequired,
        addLtl: React.PropTypes.func.isRequired,
        changeFormula: React.PropTypes.func.isRequired
	};

	render() {
		let _this = this;
		if ((_this.props.states.length != 0) && this.selected == ''){
		    this.selected = _this.props.states[0].stateName;
        }
		return (
			<div className="tl-box">
                <input className="tl-expr"
                       onChange={_this.changeFormula}
                       value={_this.props.formula}
                placeholder="TL expression"/>
                <div className="boolean-operators-box">
                    {
                        _this.binary.map((b) => {
                            return (
                                <button className="btn-boolean"
                                        key={b}
                                        value={b}
                                        onClick={_this.onAppend}>
                                    {b}
                                </button>
                            )
                        })
                    }
                </div>
                <div className="ltl-operators-box">
                    {
                        _this.ctl.map((b) => {
                            return (
                                <button className="btn-ctl"
                                        key={b}
                                        value={b}
                                        onClick={_this.onAppend}>
                                    {b}
                                </button>
                            )
                        })
                    }
                </div>
                <div className="state-insert-box">
                    <select className="state-select" onChange={_this.changeState}>
                        {
                            _this.props.states.map( (st) => {
                                return (
                                    <option key={st.stateName} value={st.stateName}>
                                        {st.stateName}
                                    </option>
                                )
                            })
                        }
                    </select>
                    <button className="btn-add-state-tl"
                            onClick={_this.addState}>
                        Add
                    </button>
                </div>
                <button className="btn-add-tl-expr" onClick={this.addLtl}>
                    Add
                </button>
            </div>
		);
	}

	addLtl(){
	    this.props.addLtl(this.props.formula);
	    this.props.changeFormula('');
    }

	addState() {
        this.props.changeFormula(this.props.formula.concat(this.selected+' '));
	}


    changeFormula(event) {
        this.props.changeFormula(event.target.value);
    }

    onAppend(event) {
        this.props.changeFormula(this.props.formula.concat(event.target.value+' '));
    }

    changeState(event) {
        this.selected = event.target.value;
    }
}

// function mapStateToProps (state) {
// 	return {
// 		formula: state.CtlBoxReducer.formula
// 	}
// }
//
// function mapDispatchToProps(dispatch) {
// 	return {
// 		changeFormula: bindActionCreators(changeFormula, dispatch)
// 	}
// }
//
// export default connect(mapStateToProps, mapDispatchToProps)(CtlBox)
