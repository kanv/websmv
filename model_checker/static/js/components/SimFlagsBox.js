import React from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import changeSourceCode from '../actions/SourceCodeEditorActions'

export class SimFlagsBox extends React.Component {

	static propTypes = {
        simFlags: React.PropTypes.object.isRequired,
        changeMode: React.PropTypes.func.isRequired
	};

	render() {
		let _this = this;
		return (
			<div>
            </div>
		);
	}
}