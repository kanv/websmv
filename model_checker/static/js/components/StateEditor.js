import React from 'react'

export class StateEditorItem extends React.Component {
    constructor(props) {
        super(props);
        this.enumItem = '';
        this.name = '';
        this.range = this.props.stateItem.range;
        this.onAdd = this.onAdd.bind(this);
        this.onChangeName = this.onChangeName.bind(this);
        this.onChangeType = this.onChangeType.bind(this);
        this.onAddEnum = this.onAddEnum.bind(this);
        this.onRemoveEnum = this.onRemoveEnum.bind(this);
        this.onEnumItemChange = this.onEnumItemChange.bind(this);
        this.onChangeType = this.onChangeType.bind(this);
        this.onChangeRangeLow = this.onChangeRangeLow.bind(this);
        this.onChangeRangeHigh = this.onChangeRangeHigh.bind(this);
        this.onChangeArrayType = this.onChangeArrayType.bind(this);
        this.reset = this.reset.bind(this);
    }

    static propTypes = {
        stateTypes:React.PropTypes.array.isRequired,
        stateItem:React.PropTypes.object.isRequired,
        stateEditorActions: React.PropTypes.func.isRequired,
        changeType: React.PropTypes.func.isRequired,
        changeName:React.PropTypes.func.isRequired,
        addEnum:React.PropTypes.func.isRequired,
        removeEnum:React.PropTypes.func.isRequired,
        setRange:React.PropTypes.func.isRequired,
        setArrayType:React.PropTypes.func.isRequired,
        reset:React.PropTypes.func.isRequired,
		addState:React.PropTypes.func.isRequired
    };

    onChangeName(e){
        this.props.changeName(e.target.value);
    }

    onChangeType(e) {
        this.props.changeType(e.target.value);
    }

    onAdd(e) {
		this.props.addState(this.props.stateItem);
        this.props.stateEditorActions(this.props.stateItem);
        this.reset();
    }

    reset(){
        this.props.reset();
        this.props.changeType('bool');
        this.range = [0,0];
        this.props.setRange(this.range);
        this.enumItem = '';
    }

    onAddEnum(e) {
      this.props.addEnum(this.enumItem);
    }

    onRemoveEnum(e) {
      this.props.removeEnum(e.target.value);
    }

    onEnumItemChange(e) {
      this.enumItem = e.target.value;
    }

    onChangeRangeLow(e){
      this.range[0] = e.target.value;
      this.props.setRange(this.range);
    }

    onChangeRangeHigh(e){
      this.range[1] = e.target.value;
      this.props.setRange(this.range);
    }

    onChangeArrayType(e) {
      this.props.setArrayType(e.target.value);
    }

    render() {
        let _this = this;
        switch(_this.props.stateItem.type){
          case 'bool':{
          return (
              <div className="state-item">
                    <div className="state-name-edit">
                      <input type="text" onChange={_this.onChangeName}
                          value={_this.props.stateItem.stateName}
                             placeholder="Var's name"/>
                      <span className="highlight"></span>
                      <span className="bar"></span>
                    </div>
                      <select className="state-type-edit"
					  		onChange={_this.onChangeType}
							value="bool">
                          <option value="bool">Bool</option>
                          <option value="enum">Enum</option>
                          <option value="array">Array</option>
                      </select>
                  <div className="round-button">
                  <div className="round-button-circle">
                      <a href="#" className="round-button"
                      onClick={_this.onAdd}>
                      +
                  </a></div></div>
              </div>
          );
        }
          case 'enum':{
            return (
              <div className="state-item">
                  <input type="name" className="state-name-edit" onChange={_this.onChangeName}
                      value={_this.props.stateItem.stateName}
                      placeholder="Var's name..."/>
				  <select className="state-type-edit"
						onChange={_this.onChangeType}
						value="enum">
					  <option value="bool">Bool</option>
					  <option value="enum">Enum</option>
					  <option value="array">Array</option>
				  </select>
              <div>
              {
                _this.props.stateItem.range.map((d)=>{
                  return (<div className="enum-item-edit" key={d}>
                  <div className="enum-item">{d}</div>
                  <div className="btn-add-enum">
                      <a  href="#"
                  value={d}
                         onClick={_this.onRemoveEnum}>-</a></div>
                  <div className="separator"/>
                  </div>
                );
              })
            }
            <input placeholder="Enumeration item..." className="enum-item-add" onChange={_this.onEnumItemChange}/>
            <div className="btn-add-enum" >
                <a href="#" onClick={_this.onAddEnum}>+</a></div>
            </div>
            <div className="round-button">
                  <div className="round-button-circle">
                      <a href="#" className="round-button"
                      onClick={_this.onAdd}>
                      +
                  </a></div></div>
              </div>
          );
          }
          case 'array':{
          return (
            <div className="state-item">
                  <input type="name" className="state-name-edit" onChange={_this.onChangeName}
                      value={_this.props.stateItem.stateName}
                      placeholder="Var's name..."/>
				  <select className="state-type-edit"
						onChange={_this.onChangeType}
						value="array">
					  <option value="bool">Bool</option>
					  <option value="enum">Enum</option>
					  <option value="array">Array</option>
				  </select>
                  <div>
                  <div className="label-state-range">Index from</div>
                  <input type="name" className="state-range-edit" onChange={_this.onChangeRangeLow}
                  value={_this.props.stateItem.range[0]}
                  placeholder="Index from..."/>
                  <div className="label-state-range">Index to</div>
                  <input type="name" className="state-range-edit" onChange={_this.onChangeRangeHigh}
                  value={_this.props.stateItem.range[1]}
                  placeholder="Index to..."/>
                    <select className="state-type-edit-arr" onChange={_this.onChangeArrayType}>
                      {
                        _this.props.stateTypes.map((d)=>{
                          return (
                                <option key={d} value={d}>{d}</option>
                              );
                        })
                      }
                      </select>
                  </div>
                <div className="round-button">
                  <div className="round-button-circle">
                      <a href="#" className="round-button"
                      onClick={_this.onAdd}>
                      +
                  </a></div></div>
                </div>
          );
        }
        }
    }
}
