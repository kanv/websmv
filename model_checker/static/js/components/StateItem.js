import React from 'react'
import findObjInArray from '../util/ArrayAdditions'

export class StateItem extends React.Component {
    constructor(props) {
        super(props);
        this.select = [];
        this.onRemove = this.onRemove.bind(this);
        this.onChangeSelect = this.onChangeSelect.bind(this);
        this.onChangeSelectArr = this.onChangeSelectArr.bind(this);
    }

    static propTypes = {
        state: React.PropTypes.object.isRequired,
        vertex: React.PropTypes.object.isRequired,
        enums: React.PropTypes.array.isRequired,
        removeState: React.PropTypes.func.isRequired,
        changeSelect: React.PropTypes.func.isRequired,
        changeSelectArr: React.PropTypes.func.isRequired
    };

    onRemove(event) {
        this.props.removeState({stateName: this.props.state.stateName, type: this.props.state.type});
    }

    onChangeSelect(e) {
        this.props.changeSelect(this.props.vertex,
			this.props.state.stateName, e.target.value);
    }

    onChangeSelectArr(e){
		this.props.changeSelectArr(this.props.vertex,
			this.props.state, e.target.id, e.target.value);
    }

    render() {
        let _this = this;
        let type, arrayLen, select = [];
        if (_this.props.state.type == 'array'){
        	arrayLen = _this.props.state.range[1] - _this.props.state.range[0];
        	switch (_this.props.state.arrayType){
				case 'bool':{
					for(let i = 0; i < arrayLen; i++) {
						_this.select.push(
						<li key={i}>
                        <div className="arr-select-container">
                        <div className="arr-select-label">Item {i}</div>
						<select id={i} className="state-edit-arr" onChange={_this.onChangeSelectArr}
						value={_this.props.state.value[i + _this.props.state.range[0]]}>
						<option value="TRUE">TRUE</option>
						<option value="FALSE">FALSE</option>
						</select>
                        </div>
						</li>);
					}
					break;
				}
				default:{
					let index = findObjInArray(_this.props.enums,
							_this.props.state.arrayType, 'stateName');
					if (index != -1){
						let state = _this.props.enums[index];
						for(let i = 0; i < arrayLen; i++) {
							_this.select.push(
							<li key={i}>
                            <div className="arr-select-container">
                                <div className="arr-select-label">Item {i}</div>
							<select key={'sel'+i} id={i} className="state-edit-arr" onChange={_this.onChangeSelectArr}
							value={state.value[i]}>
							{
								state.range.map((it)=>{
									return (
										<option key={'arrSel'+it} value={it}>{it}</option>
									);
								})
							}
							</select>
                            </div>
							</li>);
						}
					}
				}
        	}
        }
        switch (_this.props.state.type){
            case 'bool': {
                type = 'Boolean';
                return (
                  <div className="state-item">
                      <div type="state-name" className="state-name" value={_this.props.state.stateName}>{_this.props.state.stateName}</div>
                      <div className="state-type" >{type}</div>
                      <div className="label-var-value">Value</div>
                      <select className="state-edit" onChange={_this.onChangeSelect}
                              value={_this.props.state.value}>
                          <option value="TRUE">TRUE</option>
                          <option value="FALSE">FALSE</option>
                      </select>
                      <button className="state-remove" onClick={_this.onRemove}>-</button>
                  </div>
                )
            }
            break;
            case 'enum': {
                type = 'Enum';
                return (
                  <div className="state-item">
                      <label type="state-name" className="state-name" value={_this.props.state.stateName}>{_this.props.state.stateName}</label>
                          <label className="state-type" >{type}</label>
                      <select className="state-edit" onChange={_this.onChangeSelect}
                              value={_this.props.state.value}>
                          {
                            _this.props.state.range.map((val)=> {
                              return (
                                <option value={val}>{val}</option>
                              )
                            })
                          }
                      </select>
                          <button className="state-remove" onClick={_this.onRemove}>-</button>
                  </div>
                )
            }
            break;
            case 'array': {
                type = 'Array';
                return (
                  <div className="state-item">
                      <label type="state-name" className="state-name" value={_this.props.state.stateName}>{_this.props.state.stateName}</label>
                          <label className="state-type" >{type}</label>
                      <ul className="array-select-container">
                          {_this.select}
                      </ul>
                          <button className="state-remove" onClick={_this.onRemove}>-</button>
                  </div>
            );}
            break;
        }
    }
}
