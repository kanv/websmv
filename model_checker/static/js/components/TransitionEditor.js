import React from 'react'
import findObjInArray from '../util/ArrayAdditions';

export class TransitionEditor extends React.Component {
    constructor(props) {
        super(props);
		this.select = {};
		this.selectArr = [];
		this.arrVals = [];
		this.onChangeStateName = this.onChangeStateName.bind(this);
		this.onChangeArrValue = this.onChangeArrValue.bind(this);
		this.onChangeValue = this.onChangeValue.bind(this);
		this.onAddTransition = this.onAddTransition.bind(this);
    }

	componentWillMount(){
		this.props.reset();

	}

    static propTypes = {
        states:React.PropTypes.array.isRequired,
        edge:React.PropTypes.object.isRequired,
        enums:React.PropTypes.array.isRequired,
		transition:React.PropTypes.object.isRequired,
		changeName:React.PropTypes.func.isRequired,
		changeValue:React.PropTypes.func.isRequired,
		addTransition:React.PropTypes.func.isRequired,
		reset:React.PropTypes.func.isRequired,
    };

	onChangeStateName(e){
		this.props.changeName(e.target.value);
		let name = this.props.transition.name;
		let index = findObjInArray(this.props.states, name, 'stateName');
		if (index != -1 && this.props.states[index].type == 'array') {
			let state = {...this.props.states[index]};
			let i = findObjInArray(_this.props.enums,
				state.arrayType, 'stateName');
			if (i != -1){
				let enumm = _this.props.enums[index];
				this.arrVals = [];
				for(let j = state.range[0]; j < state.range[1]; j++){
					this.arrVals.push(enumm.range[0]);
				}
			}
		}
	}

	onChangeValue(e){
		this.props.changeValue(e.target.value);
	}

	onChangeArrValue(e){
		this.arrVals[e.target.id] = e.target.value;
		this.props.changeValue(this.arrVals);
	}

	onAddTransition(e){
		this.props.addTransition(this.props.edge, this.props.transition);
		this.props.reset();
		this.arrVals = [];
	}

    render() {
		let _this = this;
		_this.select = <select onChange={_this.onChangeStateName}
					value={_this.props.transition.name}>
					{
						_this.props.states.map((st)=>{
							return(
								<option key={st.stateName} value={st.stateName}>
								{st.stateName}
								</option>
							);
						})
					}
					</select>
		let name = _this.props.transition.name;
		let index = findObjInArray(_this.props.states, name, 'stateName');
		if (index != -1) {
			let state = {..._this.props.states[index]};
			if(state.type == 'array'){
				arrayLen = state.range[1] - state.range[0];
	        	switch (_this.props.state.arrayType){
					case 'bool':{
						for(let i = 0; i < arrayLen; i++) {
							_this.arrVals[i] = 'false';
							_this.selectArr.push(
							<li key={i}>
							<label>{parseInt(state.range[0]) + i}</label>
							<select id={i} className="state-edit-arr" onChange={_this.onChangeArrValue}
							value="true">
							<option value="true">TRUE</option>
							<option value="false">FALSE</option>
							</select>
							</li>);
						}
						break;
					}
					default:{
						let index = findObjInArray(_this.props.enums,
								state.arrayType, 'stateName');
						if (index != -1){
							let enumm = _this.props.enums[index];
							for(let i = 0; i < arrayLen; i++) {
								_this.selectArr.push(
								<li key={i}>
								<label>{parseInt(state.range[0]) + i}</label>
								<select key={'selTr'+i} id={i} className="state-edit-arr" onChange={_this.onChangeSelectArr}
								value={enumm.range[0]}>
								{
									enumm.range.map((it)=>{
										return (
											<option key={'arrSel'+it} value={it}>{it}</option>
										);
									})
								}
								</select>
								</li>);
							}
						}
					}
	        	}
			}
			switch(state.type){
				case 'bool':{
					return(
						<div>
						{
							_this.select
						}
						<select onChange={_this.onChangeValue}
								value="true">
							<option value="true">TRUE</option>
							<option value="false">FALSE</option>
						</select>
						<button onClick={_this.onAddTransition}>+</button>
						</div>
					)
				}
				case 'enum':{
					return(
						<div>
						{
							_this.select
						}
						<select onChange={_this.onChangeValue}
								value={state.range[0]}>
						{
							state.range.map((r)=>{
								return (
									<option key={'tr'+r} value={r}>{r}</option>
								)
							})
						}
						</select>
						<button onClick={_this.onAddTransition}>+</button>
						</div>
					)
				}
				case 'array':{
					return (
						<div>
						{
							_this.select
						}
						<ul className="array-select-container">
							{_this.selectArr}
						</ul>
						<button onClick={_this.onAddTransition}>+</button>
	                  </div>)
				}
			}
		} else return(<div/>)
	}
}
