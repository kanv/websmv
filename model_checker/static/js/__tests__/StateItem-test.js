import React from 'react';
import {StateItem} from '../components/StateItem';
import renderer from 'react-test-renderer';

test('Select value change test', () => {

    const state = {stateName: 'test', type: 'bool', value: 'false'},
        vertex = {},
        enums = [],
        removeState = jest.fn(),
        changeSelect = jest.fn(),
        changeSelectArr = jest.fn();

    const component = renderer.create(
        <StateItem
            state={state}
            vertex={vertex}
            enums={enums}
            removeState={removeState}
            changeSelect={changeSelect}
            changeSelectArr={changeSelectArr} />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onChange();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});