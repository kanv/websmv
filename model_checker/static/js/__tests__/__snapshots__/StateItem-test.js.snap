// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select value change test 1`] = `
<div
  className="state-item"
>
  <div
    className="state-name"
    type="state-name"
    value="test"
  >
    test
  </div>
  <div
    className="state-type"
  >
    Boolean
  </div>
  <div
    className="label-var-value"
  >
    Value
  </div>
  <select
    className="state-edit"
    onChange={[Function]}
    value="false"
  >
    <option
      value="TRUE"
    >
      TRUE
    </option>
    <option
      value="FALSE"
    >
      FALSE
    </option>
  </select>
  <button
    className="state-remove"
    onClick={[Function]}
  >
    -
  </button>
</div>
`;
